### PR TITLE
feat(config): add per-registry URL overrides for private mirrors

### PR DESCRIPTION
**Pull request by DisturbedOcean**
- Add optional `url` field to `RegistryConfig` for all registries (npm, crates, goProxy, pypi, github, jsr, pnpmCatalog)

- Add `DockerRegistryConfig` with four independent URL overrides (dockerHubRegistryUrl, dockerHubAuthUrl, ghcrRegistryUrl, ghcrAuthUrl)

- Implement custom `Debug` on both config types that redacts `user:password@` userinfo from URLs to prevent credential leakage in logs

- Add `redact_userinfo()` helper with focused unit tests covering edge cases (path-only `@`, no credentials, scheme-less input)

- Rename `create_default_resolvers` → `create_resolvers(config)` so resolvers are built from live config; keep `create_default_resolvers` as a zero-arg convenience wrapper

- Wrap `Backend::resolvers` in `Arc<RwLock<...>>` and rebuild the resolver map on every `workspace/configuration` push so URL overrides take effect without restart

- Add `DockerRegistry::with_overrides()` constructor that falls back to hardcoded defaults for any `None` argument

- Switch Rust toolchain pin from `1.92.0` to `stable`

- Update README and ARCHITECTURE.md with URL override configuration reference and credential embedding guidance

### CODE REVIEW

**Comment by skanehira** [comment]
@DisturbedOcean 
Thank you for your contribution!
My work is a bit busy, so I'll take a look at the PR later.

**Comment by DisturbedOcean** [comment]
Great. I have a follow on change to add support for Swift Package Manager

File: src/lsp/backend.rs [hunk 1]
```diff
@@ -459,26 +476,24 @@ impl<S: VersionStorer> LanguageServer for Backend<S> {
...
+                return Ok(None);
+            };
+            resolver.matcher().clone()
         };
 
         // For GitHub Actions with commit hash, use async function to fetch SHA
         let mut actions = if package.registry_type == RegistryType::GitHubActions
             && package.commit_hash.is_some()
         {
             let sha_fetcher = GitHubRegistry::default();
```
Comment (skanehira) [inline src/lsp/backend.rs]: [Critical] L492: code_action path doesn't respect the `github.url` override

This builds a fresh `GitHubRegistry::default()` on every code action invocation, so even when the user overrides `registries.github.url` in the LSP config, the commit-hash → SHA fetch path keeps falling back to public github.com (or the `GITHUB_API_BASE_URL` env var).

For GitHub Enterprise or auth-protected mirrors, the SHA fetch issued from a code action will fail, which breaks the very motivation of this PR (private mirror support).

Suggested fixes:
1. Reuse the GitHub Actions registry from `resolvers` — either add a `sha_fetcher()` accessor to `PackageResolver`, or implement `TagShaFetcher` on `GitHubRegistry` so the same instance can be used here.
2. As a minimal fix, read `github.url` from the live config and construct `GitHubRegistry::new(url)` at this call site.

(1) is cleaner since it consolidates `GitHubRegistry::default()` usage in the resolver layer and keeps the override logic in one place.

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.92.0"
+channel = "stable"
```
Comment (skanehira) [inline rust-toolchain.toml]: Since we maintain a fixed compiler version and regularly perform manual updates, I'd like to request that this change be reverted.

Comment (DisturbedOcean) [reply rust-toolchain.toml]: Reverted

File: src/lsp/backend.rs (hunk 1 above)
Comment (DisturbedOcean) [reply src/lsp/backend.rs]: Updated

**Comment by skanehira** [comment]
@DisturbedOcean 
The CI is failing, but I suspect it's fixed in the main branch. Could you merge the main branch into our working branch?

**Comment by DisturbedOcean** [comment]
> @DisturbedOcean The CI is failing, but I suspect it's fixed in the main branch. Could you merge the main branch into our working branch?

Done

**Comment by skanehira** [comment]
@DisturbedOcean Thank you :)
